pkg/clients/aws.go

```diff
@@ -102,12 +105,35 @@ func UseProviderConfig(ctx context.Context, c client.Client, mg resource.Managed
 		if err := c.Get(ctx, types.NamespacedName{Namespace: csr.Namespace, Name: csr.Name}, s); err != nil {
 			return nil, errors.Wrap(err, "cannot get credentials secret")
 		}
-		return UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		cfg, err := UseProviderSecret(ctx, s.Data[csr.Key], DefaultSection, region)
+		return SetResolver(ctx, mg, cfg), err
 	default:
 		return nil, errors.Errorf("credentials source %s is not currently supported", s)
 	}
 }
 
+// SetResolver parses annotations from the managed resource
+// and returns a configuration accordingly.
+func SetResolver(ctx context.Context, mg resource.Managed, cfg *aws.Config) *aws.Config {
+	if URL, ok := mg.GetAnnotations()["aws.crossplane.io/endpointURL"]; ok {
+		if SigningRegion, ok := mg.GetAnnotations()["aws.crossplane.io/endpointSigningRegion"]; ok {
+			defaultResolver := endpoints.NewDefaultResolver()
+			endpointResolver := func(service, region string) (aws.Endpoint, error) {
+				if strings.Contains(mg.GetResourceVersion(), service) {
```

Comment:
I believe `GetResourceVersion` will return the `metadata.resourceVersion` of the managed resource, which is like:

```yaml
metadata:
  resourceVersion: "147443"
```

Is using this field intentional? It doesn't seem like it would ever contain the `service` name.

Reply:
Ah I'm trying to get the `apiVersion` field.

Reply:
Is it `mg.GetObjectKind().GroupVersionKind().String()`